docs/tutorials/tutorial-update-data.md

```diff
@@ -23,161 +23,236 @@ sidebar_label: Update existing data
   ~ under the License.
   -->
 
+:::info
 
-This tutorial shows you how to update data in a datasource by overwriting existing data and adding new data to the datasource.
+Apache Druid supports two query languages: [Druid SQL](../querying/sql.md) and [native queries](../querying/querying.md).
+This document describes the SQL language.
+
+:::
+
+Apache Druid stores data and indexes in [segment files](../design/segments.md) partitioned by time.
+Once segments are created, they cannot be modified.
+To update existing data, you must rebuild and republish the segments.
+Druid updates existing data using time ranges, not primary key. Data outside the replacement time range is not touched.
```

Comment:
```suggestion
In Druid use time ranges to specify data to update, not primary key or dimensions like you may be familiar with in transactional databases. Data outside the specified replacement time range is not touched.
```

Reply:
I would also add here:  You can use this functionality in Druid to perform data updates, inserts, and deletes, similar to UPSERT functionality for transactional databases.

(we need to have UPSERT mentioned somewhere)

Reply:
Updated text